match/tensordata.py

```diff
@@ -124,7 +127,7 @@ def __multi_to_single_rank_translation(self, coords: tuple) -> int:
         self.__out_of_bounds_coords(coords)
         return int(sum(dim * stride for dim, stride in zip(coords, self._strides)))
 
-    def item(self):
+    def item(self) -> Union[int, float]:
```

Comment:
Use the newer syntax if possible (I think it should be with 3.11)

```suggestion
    def item(self) -> int | float:
```